Content.Client/Atmos/Monitor/UI/Widgets/ScrubberControl.xaml.cs

```diff
@@ -0,0 +1,126 @@
+    public partial class ScrubberControl : BoxContainer
+    {
+        private GasVentScrubberData _data;
+        private string _address;
+
+        public event Action<string, IAtmosDeviceData>? ScrubberDataChanged;
+
+        private CheckBox _enabled => CEnableDevice;
+        private CollapsibleHeading _addressLabel => CAddress;
+        private OptionButton _pumpDirection => CPumpDirection;
+        private FloatSpinBox _volumeRate => CVolumeRate;
+        private CheckBox _wideNet => CWideNet;
+
+        private GridContainer _gases => CGasContainer;
+        private Dictionary<Gas, Button> _gasControls = new();
+
+        public ScrubberControl(GasVentScrubberData data, string address)
+        {
+            RobustXamlLoader.Load(this);
+
+            this.Name = address;
+
+            _data = data;
+            _address = address;
+
+            _addressLabel.Title = Loc.GetString("air-alarm-ui-atmos-net-device-label", ("address", $"{address}"));
+
+            _enabled.Pressed = data.Enabled;
+            _enabled.OnToggled += _ =>
+            {
+                _data.Enabled = _enabled.Pressed;
+                ScrubberDataChanged?.Invoke(_address, _data);
+            };
+
+            _wideNet.Pressed = data.WideNet;
+            _wideNet.OnToggled += _ =>
+            {
+                _data.WideNet = _wideNet.Pressed;
+                ScrubberDataChanged?.Invoke(_address, _data);
+            };
+
+            _volumeRate.Value = (float) _data.VolumeRate!;
+            _volumeRate.OnValueChanged += _ =>
+            {
+                _data.VolumeRate = _volumeRate.Value;
+                ScrubberDataChanged?.Invoke(_address, _data);
+            };
+            _volumeRate.IsValid += value =>
+            {
+                if (value < 0) return false;
+
+                return true;
+            };
+
+            foreach (var value in Enum.GetValues<ScrubberPumpDirection>())
+                _pumpDirection.AddItem(Loc.GetString($"{value}"), (int) value);
+
+            _pumpDirection.SelectId((int) _data.PumpDirection!);
+            _pumpDirection.OnItemSelected += args =>
+            {
+                _pumpDirection.SelectId(args.Id);
+                _data.PumpDirection = (ScrubberPumpDirection) args.Id;
+                ScrubberDataChanged?.Invoke(_address, _data);
+            };
+
+            foreach (var value in Enum.GetValues<Gas>())
+            {
+                var gasButton = new Button
+                {
+                    Name = value.ToString(),
+                    Text = Loc.GetString($"{value}"),
+                    ToggleMode = true,
+                    HorizontalExpand = true,
+                    Pressed = _data.FilterGases!.Contains(value)
+                };
+                gasButton.OnToggled += args =>
+                {
+                    if (args.Pressed) _data.FilterGases.Add(value);
+                    else              _data.FilterGases.Remove(value);
```

Comment:
bro what

I don't like the C# styleguide either, but like
dude

```suggestion
                    if (args.Pressed)
                        _data.FilterGases.Add(value);
                    else
                        _data.FilterGases.Remove(value);
```